### PR TITLE
Use _six for inf and nan

### PR DESCRIPTION
**Pull request by ssnl**
Things like `float('inf')` are actually quite expensive.
```py
In [1]: import math

In [2]: %timeit -n 200 math.inf
49.3 ns ± 1.42 ns per loop (mean ± std. dev. of 7 runs, 200 loops each)

In [3]: %timeit -n 200 float('inf')
194 ns ± 39.1 ns per loop (mean ± std. dev. of 7 runs, 200 loops each)
```

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by SsnL** [comment]
i'll merge after #9487 is merged.

**Pull request closed without merging**
